### PR TITLE
Remove skipifs from streamio tests

### LINKED ISSUE #1534: Segfaults in utils/test_streamio.py
As seen in #1472 there are segfaults if we try to run some tests in `test_streamio.py`. The current fix is to just mark those tests for skipping.

The travis log for the [old numpy build](https://travis-ci.org/MDAnalysis/mdanalysis/jobs/252952480)

```
____________________ MDAnalysisTests/utils/test_streamio.py ____________________
[gw0] linux2 -- Python 2.7.13 /home/travis/miniconda/envs/test/bin/python
Slave 'gw0' crashed while running 'MDAnalysisTests/utils/test_streamio.py::TestNamedStream_filename_behavior::()::test_expanduser_noexpansion_returns_NamedStream'
____________________ MDAnalysisTests/utils/test_streamio.py ____________________
[gw2] linux2 -- Python 2.7.13 /home/travis/miniconda/envs/test/bin/python
Slave 'gw2' crashed while running 'MDAnalysisTests/utils/test_streamio.py::TestNamedStream_filename_behavior::()::test_expandvars'
____________________ MDAnalysisTests/utils/test_streamio.py ____________________
[gw3] linux2 -- Python 2.7.13 /home/travis/miniconda/envs/test/bin/python
Slave 'gw3' crashed while running 'MDAnalysisTests/utils/test_streamio.py::TestNamedStream_filename_behavior::()::test_expandvars_noexpansion_returns_NamedStream'
============== 3 failed, 5123 passed, 1 skipped in 560.61 seconds ==============
```

There is also a comment in the tests by @orbeckst `Segmentation fault when run as a test on Mac OS X 10.6, Py 2.7.11 [orbeckst]`

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #1534

Changes made in this Pull Request:
 - Issue #1534 details some tests which had to be skipped due to segfaults. Testing locally I can no longer reproduce this issue (although the tests themselves were broken and needed fixing). Here I'm attempting to see if it works with CI, if so, we should just go ahead and bring them back?

PR Checklist
------------
 - [x] Tests?
 - Docs?
 - CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: testsuite/MDAnalysisTests/utils/test_streamio.py [hunk 1]
```diff
@@ -259,8 +255,10 @@ class _StreamData(object):
     }
 
     def __init__(self):
-        self.buffers = {name: "".join(open(fn).readlines())
-                        for name, fn in self.filenames.items()}
+        self.buffers = {}
```
Comment (IAlibay) [inline testsuite/MDAnalysisTests/utils/test_streamio.py:258]: also did a little bit of cleaning to remove some of the excess warnings being raised - this originally left a bunch of open file handles so I changed it to use context managers.

**Review by tylerjereddy — COMMENTED** [review]
Looks reasonable at first glance. I usually see warnings filters in i.e., `pytest.ini`--you have a need to scope them this narrowly?

**Comment by IAlibay** [comment]
> Looks reasonable at first glance. I usually see warnings filters in i.e., `pytest.ini`--you have a need to scope them this narrowly?

Probably not - I remember we did something similar for another tests in response to https://github.com/MDAnalysis/mdanalysis/issues/2980. If it doesn't affect warnings catching I guess we could just apply some global filters?

**Comment by IAlibay** [comment]
I _think_ that should work? Let's see how it does - number of warnings to beat is 47032 

Reduced to 45965, that's not a bad outcome!

**Review by richardjgowers — APPROVED** [review]
Lgtm

**Pull request merged**